packages/cdktf-cli/lib/get/generator/emitter/attributes-emitter.ts

```diff
@@ -12,217 +20,132 @@ export class AttributesEmitter {
-    this.code.openBlock(`public get ${att.name}()`);
-    this.code.line(
-      `return ${
-        att.isProvider
-          ? "this." + att.storageName
-          : this.determineGetAttCall(att)
-      };`
-    );
-    this.code.closeBlock();
-
-    this.code.openBlock(`public set ${att.name}(value: ${att.type.name})`);
-    this.code.line(`this.${att.storageName} = value;`);
-    this.code.closeBlock();
-
-    this.code.line(`// Temporarily expose input value. Use with caution.`);
-    this.code.openBlock(`public get ${this.getInputName(att, escapeInput)}()`);
-    this.code.line(`return this.${att.storageName}`);
-    this.code.closeBlock();
-  }
-
-  private emitComputedComplexList(att: AttributeModel) {
-    this.code.openBlock(`public ${att.name}(index: string)`);
-    this.code.line(
-      `return new ${att.type.name}(this, '${att.terraformName}', index);`
-    );
-    this.code.closeBlock();
-  }
 
-  private emitComputedComplexListMap(att: AttributeModel) {
-    this.code.openBlock(
-      `public ${att.name}(index: string, key: string): ${this.determineMapType(
-        att
-      )}`
-    );
-    this.code.line(
-      `return new ${att.type.name}(this, \`${att.terraformName}.\${index}\`).lookup(key);`
-    );
-    this.code.closeBlock();
-  }
+    if (isStored) {
+      this.code.line(
+        `private ${att.storageName}${att.isOptional ? "?" : ""}: ${
+          att.type.name
+        };`
+      );
+    }
 
-  private emitComputedComplexOptional(
-    att: AttributeModel,
-    escapeReset: boolean,
-    escapeInput: boolean
-  ) {
-    this.code.line(`private ${att.storageName}?: ${att.type.name}`);
-    this.code.openBlock(`public get ${att.name}(): ${att.type.name}`);
-    this.code.line(
-      `return this.interpolationForAttribute('${att.terraformName}') as any; // Getting the computed value is not yet implemented`
-    );
-    this.code.closeBlock();
+    switch (getterType._type) {
+      case "getter":
+        this.code.openBlock(`public get ${att.name}()`);
+        this.code.line(`return ${this.determineGetAttCall(att)}`);
+        this.code.closeBlock();
+        break;
 
-    this.code.openBlock(`public set ${att.name}(value: ${att.type.name})`);
-    this.code.line(`this.${att.storageName} = value;`);
-    this.code.closeBlock();
+      case "function":
+        this.code.openBlock(
+          `public ${att.name}(${getterType.args}): ${
+            getterType.returnType || att.type.name
+          }`
+        );
+        this.code.line(`return ${getterType.returnStatement};`);
+        this.code.closeBlock();
+        break;
+    }
 
-    this.code.openBlock(`public ${this.getResetName(att.name, escapeReset)}()`);
-    this.code.line(`this.${att.storageName} = undefined;`);
-    this.code.closeBlock();
+    if (hasSetter) {
+      this.code.openBlock(
+        `public set ${att.name}(value: ${att.type.name}${
+          hasResetMethod ? " | undefined" : ""
```

Comment:
This doesn't match the existing behavior which only allowed `undefined` for optional attributes of a provider. See https://github.com/hashicorp/terraform-cdk/pull/310 for a discussion on the reasoning behind what was implemented.